core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -370,22 +362,21 @@ private int binaryFind(K key) {
         int sortedCount = this.sortedCount.get();
         // if there are no sorted keys, or the first item is already larger than key -
         // return the head node for a regular linear search
-        if ((sortedCount == 0) || compare(readKey(FIRST_ITEM), key) >= 0) {
+        if ((sortedCount == 0) || comparator.compareKeyAndSerializedKey(key, readKey(FIRST_ITEM)) <= 0) {
```

Comment:
The opposite of ">=" is "<" and not "<=". Also in couple of lines bellow "<" was changed to ">", while the opposite of "<" is ">=". Are we sure that the condition of keys being equal are treated correctly here?

Reply:
Please verify it in all lines where comparator operands places where changes. As I see it in other lines also, just do not comment there as well...

Reply:
Note that we're not looking for the opposite condition. We still check the same semantic:  "Is the first item larger than or equals key". It's just that the sign is flipped because of argument order.

To make sure, I also tried running tests after replacing with exact opposites like you suggest and multiple tests fail.

Reply:
OK, I agree